src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -739,14 +742,14 @@ export class EditorGroupModel extends Disposable implements IEditorGroupModel {
 	private doSetSelection(newActiveEditor: EditorInput, activeEditorIndex: number, inactiveSelectedEditors: EditorInput[]): void {
 		this.doSetActive(newActiveEditor, activeEditorIndex);
 
-		for (const candidate of inactiveSelectedEditors) {
+		for (const candidate of inactiveSelectedEditors) { // TODO should this move up to the public API? the idea is that we do not trust editor inputs from public API but internal is fine
```

Comment:
@benibenj the idea in the model is that public methods can pass in "untrusted" editor inputs that we need to match against our model, but methods that are called internally can assume this has alreaddy happened for the editor inputs passed in. Not sure about this case, but maybe can be done.